### PR TITLE
Equiopment block editing

### PR DESCRIPTION
**Pull request by vyMoon**

### CODE REVIEW

File: src/app/admin/services/api/api.service.ts [hunk 1]
```diff
@@ -39,14 +39,18 @@ export class ApiService {
     });
   }
 
-  blockEquipment(id: number, period: Period) {
+  blockEquipment(id: number, period: Period): Observable<unknown> {
```
Comment (vitaliy-pak) [inline src/app/admin/services/api/api.service.ts:42]: return type is string observable

File: src/app/admin/services/api/api.service.ts [hunk 2]
```diff
@@ -39,14 +39,18 @@ export class ApiService {
...
     const body = {
       end_date: period.endDate,
       start_date: period.startDate,
     };
     return this.http.post<string>(`/equipment/${id}/blocking`, body);
   }
 
+  unblockEquipment(id: number): Observable<null> {
```
Comment (vitaliy-pak) [inline src/app/admin/services/api/api.service.ts:50]: I think void observable is more preferrable to indicate that code doesn't rely on the emitted value and null observable is rarely used.

File: src/app/admin/services/controller/equipment-controller.service.ts [hunk 3]
```diff
@@ -116,18 +117,37 @@ export class EquipmentController {
...
-      });
+      .subscribe();
+  }
+
+  blockEquipment(
+    equipment: Equipment,
+    period: Period,
+    unavailableDates: UnavailableDates[],
+    action: string,
+  ): Observable<unknown> {
```
Comment (vitaliy-pak) [inline src/app/admin/services/controller/equipment-controller.service.ts:136]: again as no rely on the emitted value and no clear return type, better to use void observable

File: src/app/admin/services/controller/equipment-controller.service.ts [hunk 4]
```diff
@@ -116,18 +117,37 @@ export class EquipmentController {
...
+    return of(this.isPeriodsIntersect(period, unavailableDates)).pipe(
+      switchMap((isIntersect) => (isIntersect ? this.openOrderNotificationModal(action) : of(true))),
+      filter(Boolean),
+      switchMap(() => this.api.blockEquipment(equipment.id, period)),
+      tap(() => this.notificationService.openSuccess(`${equipment.title} ${EquipmentNotification.blocked}`)),
+    );
+  }
+
+  unblockEquipment(equipment: Equipment): Observable<null> {
```
Comment (vitaliy-pak) [inline src/app/admin/services/controller/equipment-controller.service.ts:145]: same here

**Review by vitaliy-pak — CHANGES_REQUESTED** [review]

File: src/app/admin/services/api/api.service.ts (hunk 1 above)
Comment (vyMoon) [reply src/app/admin/services/api/api.service.ts]: done

File: src/app/admin/services/controller/equipment-controller.service.ts (hunk 3 above)
Comment (vyMoon) [reply src/app/admin/services/controller/equipment-controller.service.ts]: done

File: src/app/admin/services/controller/equipment-controller.service.ts (hunk 4 above)
Comment (vyMoon) [reply src/app/admin/services/controller/equipment-controller.service.ts]: done

**Pull request merged**